### PR TITLE
QA-120 Implement adcm_xx.upgrade()

### PR DESCRIPTION
**Pull request by madlabman**

### CODE REVIEW

File: src/adcm_pytest_plugin/docker_utils.py [hunk 1]
```diff
@@ -325,6 +329,32 @@ def __init__(self, container: Container, ip: str, port: str, container_config: O
     def stop(self):
         """Stops container"""
         self.container.stop()
+        with suppress(NotFound):
+            condition = "removed" if self.container.attrs["HostConfig"]["AutoRemove"] else "stopped"
+            self.container.wait(condition=condition, timeout=30)
+
+    @allure.step("Upgrade adcm to {target}")
+    def upgrade(self, target: Tuple[str, str]) -> None:
```
Comment (dgusakov) [inline src/adcm_pytest_plugin/docker_utils.py:337]: This method should raise an error if no volume was mounted to the ADCM container before.

Comment (madlabman) [reply src/adcm_pytest_plugin/docker_utils.py:337]: Done.

File: src/adcm_pytest_plugin/docker_utils.py [hunk 2]
```diff
@@ -325,6 +329,33 @@ def __init__(self, container: Container, ip: str, port: str, container_config: O
...
+        image, tag = target
+        assert self.container_config.volumes, "has no volume to move data"
```
Comment (dgusakov) [inline src/adcm_pytest_plugin/docker_utils.py:339]:
```suggestion
        assert self.container_config.volumes, "There is no volume to move data. Make sure you are using the correct ADCM fixture for upgrade"
```

File: src/adcm_pytest_plugin/docker_utils.py [hunk 3]
```diff
@@ -325,6 +329,33 @@ def __init__(self, container: Container, ip: str, port: str, container_config: O
...
+            self.container_config.image = image
+            self.container_config.tag = tag
+            volume.update({"bind": "/adcm/data"})
+            allure.attach(
+                json.dumps(asdict(self.container_config), indent=2),  # config object is not serializable
+                name="Container config for upgrade",
+                attachment_type=AttachmentType.JSON,
+            )
+        with allure.step("Start newer adcm"):
+            self.stop()
```
Comment (dgusakov) [inline src/adcm_pytest_plugin/docker_utils.py:354]: Wrap this call into the separate step "Stop old container"

File: src/adcm_pytest_plugin/docker_utils.py (hunk 2 above)
Comment (madlabman) [reply src/adcm_pytest_plugin/docker_utils.py]: Done.

File: src/adcm_pytest_plugin/docker_utils.py (hunk 3 above)
Comment (madlabman) [reply src/adcm_pytest_plugin/docker_utils.py:356]: Done.

**Review by vgorkavenko — APPROVED** [review]

**Review by dgusakov — APPROVED** [review]

**Pull request merged**